addons/l10n_eg_edi_eta/models/res_company.py

```diff
@@ -14,3 +14,13 @@ class ResCompany(models.Model):
     l10n_eg_invoicing_threshold = fields.Float('Invoicing Threshold', default=0.0,
                                                help="Threshold at which you are required to give the VAT number "
                                                     "of the customer. ")
+
+    # ------------------------------------------------------------
+    # neutralize
+    # ------------------------------------------------------------
+
+    def _neutralize(self):
+        super()._neutralize()
+        self.flush()
+        self.invalidate_cache()
+        self.env.cr.execute("UPDATE res_company SET l10n_eg_production_env=False")
```

Comment:
@mao-odoo I would set l10n_eg_client_secret to '' as well as the test environment is like a "pre-production" environment.  @bakboka Would be nice to know from your side as well as people will duplicate dbs for support, ...

Reply:
Indeed, it might be worth to also remove the secret just to make sure nothing is sent to production by accident. Furthermore, I believe the secrets are different between pre-prod and prod

Reply:
done